From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 8 Oct 2015 17:38:22 +0100
Subject: perf: Fix build on architectures without CONFIG_PERF_REGS
Forwarded: http://mid.gmane.org/1444323185.2956.260.camel@decadent.org.uk

perf currently fails to link on all architectures other than arm,
arm64 and x86:

tools/perf/libperf.a(libperf-in.o): In function `parse_regs':
tools/perf/util/parse-regs-options.c:28: undefined reference to `sample_reg_masks'
tools/perf/util/parse-regs-options.c:28: undefined reference to `sample_reg_masks'
tools/perf/util/parse-regs-options.c:45: undefined reference to `sample_reg_masks'
tools/perf/util/parse-regs-options.c:38: undefined reference to `sample_reg_masks'

Fixes: bcc84ec65ad1 ("perf record: Add ability to name registers to record")
Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Cc: Stephane Eranian <eranian@google.com>
---
 tools/perf/builtin-record.c | 2 ++
 tools/perf/util/Build       | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index 142eeb3..34dd749c 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -1082,9 +1082,11 @@ struct option __record_options[] = {
 		    "sample transaction flags (special events only)"),
 	OPT_BOOLEAN(0, "per-thread", &record.opts.target.per_thread,
 		    "use per-thread mmaps"),
+#ifdef CONFIG_PERF_REGS
 	OPT_CALLBACK_OPTARG('I', "intr-regs", &record.opts.sample_intr_regs, NULL, "any register",
 		    "sample selected machine registers on interrupt,"
 		    " use -I ? to list register names", parse_regs),
+#endif
 	OPT_BOOLEAN(0, "running-time", &record.opts.running_time,
 		    "Record running/enabled time of read (:S) events"),
 	OPT_CALLBACK('k', "clockid", &record.opts,
diff --git a/tools/perf/util/Build b/tools/perf/util/Build
index 349bc96..333b08d 100644
--- a/tools/perf/util/Build
+++ b/tools/perf/util/Build
@@ -83,7 +83,7 @@ libperf-$(CONFIG_AUXTRACE) += intel-pt-decoder/
 libperf-$(CONFIG_AUXTRACE) += intel-pt.o
 libperf-$(CONFIG_AUXTRACE) += intel-bts.o
 libperf-y += parse-branch-options.o
-libperf-y += parse-regs-options.o
+libperf-$(CONFIG_PERF_REGS) += parse-regs-options.o
 
 libperf-$(CONFIG_LIBELF) += symbol-elf.o
 libperf-$(CONFIG_LIBELF) += probe-file.o
